synapse/app/synctl.py

```diff
@@ -190,7 +218,11 @@ def main():
         if start_stop_synapse:
             stop(pidfile, "synapse.app.homeserver")
 
-        # TODO: Wait for synapse to actually shutdown before starting it again
+    # Wait for synapse to actually shutdown before starting it again
+    if action == "restart" and os.path.exists(pidfile):
```

Comment:
A few points about this section:

* As I said at https://github.com/matrix-org/synapse/pull/2020#discussion_r107878979, we need some feedback on what's going on.
* More complex deployments of synapse have a bunch of workers as well as the main synapse process (`worker.pidfile for worker in workers`).
* If `start_stop_synapse` is false, we should *not* wait for synapse to exit.